models/repo.go

```diff
@@ -1580,10 +1580,19 @@ func ChangeRepositoryName(u *User, oldRepoName, newRepoName string) (err error)
 	}
 
 	// Change repository directory name.
-	if err = os.Rename(repo.RepoPath(), RepoPath(u.Name, newRepoName)); err != nil {
+	newRepoPath := RepoPath(u.Name, newRepoName)
+	if err = os.Rename(repo.RepoPath(), newRepoPath); err != nil {
 		return fmt.Errorf("rename repository directory: %v", err)
 	}
 
+	localPath := repo.LocalCopyPath()
+	if com.IsExist(localPath) {
+		_, err := git.NewCommand("remote", "set-url", "origin", newRepoPath).RunInDir(localPath)
```

Comment:
You usually need to acquire a lock before running a `git` command in the local copy repo, although I suppose in this case it may be alright not to

Reply:
Actually, if we really wanted to be safe, we would hold the local-copy lock while renaming the bare repo and while running `git remote set-url origin ...`.

Between renaming the bare repo and running `git remote set-url origin ...`, any other threads that try to run `git` commands involving `origin` in the local copy will fail.

Reply:
thanks, you are right, I'll try to do it.